lib/ansible/galaxy/collection/__init__.py

```diff
@@ -586,6 +594,180 @@ def build_collection(collection_path, output_path, force):
+
+    :param collections: A list of collection requirements to find and remove
+    :param search_paths: A list of paths to search for the collections
+    :param no_deps: A boolean to indicate whether dangling dependences should be ignored instead of removed
+    :param force: A boolean to bypass prompting when multiple matching collections are found
+    """
+    # Find all collections in all paths to ensure the deps of collections we remove are only removed if they will become unused
+    everything = []
+    for search_path in search_paths:
+        everything.extend(find_existing_collections(search_path, fallback_metadata=True))
+
+    # Find collections and unused dependencies
+    collections_to_remove = get_matching_collections(collections, everything, force)
+    if no_deps:
+        deps_still_needed = {}
+        deps_to_remove = []
+    else:
+        deps_of_removed_collections = get_dependencies(collections_to_remove, everything)
+        deps_still_needed = get_parents(deps_of_removed_collections, collections_to_remove, everything)
+        deps_to_remove = set(deps_of_removed_collections) - set(deps_still_needed.keys())
+
+    # Remove stuff
+    if collections_to_remove:
+        display.display("Removing requested collections:")
+        for collection in collections_to_remove:
+            collection.remove()
+    else:
+        display.display("No collections to remove")
+
+    if deps_to_remove:
+        display.display("Removing unused dependencies:")
+        for dep in deps_to_remove:
+            dep.remove()
+    for dep in deps_still_needed:
+        needed_by = [to_text(needs_dep, errors='surrogate_or_strict') for needs_dep in deps_still_needed[dep]]
+        display.vvvv("Skipping removing dependency '%s' because it is still used by the installed collections: %s" % (
+            to_text(dep, errors='surrogate_or_strict'), ', '.join(needed_by))
+        )
+
+
+def get_matching_collections(requirements, installed_collections, force):
+    """Gets the list of collections that match the requirements list
+
+    :param requirements: A list of collection specifications to look for in the list of collections
+    :param installed_collections: A list of CollectionRequirement instances
+    :param force: A boolean to bypass prompting when multiple matching instances are found
+    :return: A list of CollectionRequirement instances
+    """
+    matching_collections = []
+    for requirement in requirements:
+        unfiltered_matches = []
+
+        req_name = requirement[0]
+        req_version = requirement[1]
+        b_req_name = to_bytes(req_name, errors='surrogate_or_strict')
+
+        if os.path.isfile(b_req_name) or urlparse(req_name).scheme.lower() in ['http', 'https'] or len(req_name.split('.')) != 2:
+            display.display("Skipping '%s' because it is not a valid collection name. The format namespace.name is expected." % req_name)
```

Comment:
Shouldn't this explode loudly?

Reply:
Not in my opinion. I would like `ansible-galaxy collection remove` to work with a requirements.yml (using the same file to install, update, and remove collections is very convenient when there are a lot of them) and my `--ignore-errors` got a -1. Other suggestions of how to be usable with a requirements.yml are welcome.

Reply:
Well, then it needs a better logic to retrieve metadata from the concrete artifact pointers like URL/SCM (the tmp artifacts manager I'm working on should help with that). And also `len(req_name.split('.')) != 2` is a different class of an error. If an entry is clearly an invalid identifier, it shouldn't be silently ignored.

Also, if I'm reading this correctly, the message "skipping this" is printed but there's no actual skip (`continue`?) in code.

Reply:
I could download the tar.gz/git SCM, and parse the name from that to see if it's installed. If we do that, I think `verify` and upgrading a collection should probably work the same way.

Reply:
Yes, since I'm working on a standardized way to do this already, you probably shouldn't spend effort on implementing this.

Reply:
:100: